### ISSUE TITLE
`__getitem__` does not obey `_left_index`  and `_right_index`

### ISSUE THREAD

**Issue by b4be1** [eric-wieser/numpy_ringbuffer#5]
Hi @eric-wieser, simple indexing does not work as expected in combination with `pop()`. E.g.,
```python
r = RingBuffer(capacity=4, dtype=np.int32)
r.extend((1,2,3,4))
r.pop()
print(r[-1])
>>> 4
```
One would expect to see `r[-1] = 3`. The reason is that simple indexing does not check whether requested index is within the range between `_left_index`  and `_right_index`.

**Comment by eric-wieser** [comment]
The problem here is specifically negative indices, right? Nice catch

**Comment by b4be1** [comment]
It fails with positive indices too. E.g., `r[3]` returns `4`.

**Comment by eric-wieser** [comment]
Right, so it fails for indices where `0 <= i < len(self)` is `False`. I'll throw a patch together in the next day or two, unless you want to

**Comment by b4be1** [comment]
Yes, that's right. Thanks @eric-wieser , better you do it

**Comment by xgdgsc** [comment]
Any progress on this bug?

**Comment by eric-wieser** [comment]
I had a local patch, but the timings ended up way worse as a result of fixing it. I'll try to take another look

**Comment by Alexander-Shukaev** [comment]
Plan to merge?
